agent/app/agent.go

```diff
@@ -840,7 +845,13 @@ func (agent *ecsAgent) registerContainerInstance(
 			logger.Critical("Instance registration attempt with an invalid parameter", logger.Fields{
 				field.Error: err,
 			})
-			return err
+			return terminalError{err}
+		}
+		if utils.IsAWSErrorCodeEqual(err, ecsmodel.ErrCodeClientException) {
+			logger.Critical("Instance registration attempt with client performing invalid action", logger.Fields{
+				field.Error: err,
+			})
+			return terminalError{err}
 		}
 		if _, ok := err.(apierrors.AttributeError); ok {
```

Comment:
Is it possible this is also terminal error? 
> Instance registration attempt with invalid attribute(s)

Reply:
Handling that specific case is beyond scope of this change.

However, invalidParameterException is handled above, I would not expect attributeError to fire because instance attributes are an instance parameter of the RCI call. Ecs agent should correct me here if I'm wrong

Reply:
Sounds good to me. We can focus on the cases we care about now.